cmd/libs/go2idl/client-gen/generators/generator-for-group.go

```diff
@@ -0,0 +1,158 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package generators has the generators for the set-gen utility.
+package generators
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+)
+
+// genGroup produces a file for a group client, e.g. ExtensionsClient for the extension group.
+type genGroup struct {
+	generator.DefaultGen
+	outputPackage string
+	typeToMatch   *types.Type
+	group         string
+	// types in this group
+	types   []*types.Type
+	imports *generator.ImportTracker
+}
+
+// We only want to call GenerateType() once.
+func (g *genGroup) Filter(c *generator.Context, t *types.Type) bool { return t == g.typeToMatch }
+
+func (g *genGroup) Namers(c *generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.outputPackage, g.imports),
+	}
+}
+
+func (g *genGroup) Imports(c *generator.Context) (imports []string) {
+	return append(g.imports.ImportLines(), "fmt", "strings")
+}
+
+// GenerateType makes the body of a file implementing a set for type t.
+func (g *genGroup) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
+	sw := generator.NewSnippetWriter(w, c, "$", "$")
+	m := map[interface{}]interface{}{
+		"group": g.group,
+		"Group": namer.IC(g.group),
+		"types": g.types,
+	}
+	sw.Do(groupInterfaceTemplate, m)
+	sw.Do(groupClientTemplate, m)
+	// @reviewer: I used a for loop here, instead of a loop in the template, because I don't know how to define a variable (e.g. http://stackoverflow.com/questions/17284222/in-go-templates-accessing-parent-global-pipeline-within-range) in a go2idl template.
+	for _, t := range g.types {
+		wrapper := map[interface{}]interface{}{
+			"type":  t,
+			"Group": namer.IC(g.group),
+		}
+		sw.Do(nameSpacerImplTemplate, wrapper)
+	}
+	sw.Do(newClientTemplate, m)
+	sw.Do(newClientOrDieTemplate, m)
+	sw.Do(setClientDefaultsTempalte, m)
+
+	return sw.Error()
+}
+
+// @reviewer, here I use a loop in the template, is this a good practice?
+var groupInterfaceTemplate = `
+type $.Group$Interface interface {
+    // This interface should be removed
+	// VersionInterface
+    $range .types$
+        $.Name.Name$Namespacer
+    $end$
+}
+`
+
+var groupClientTemplate = `
+// $.Group$Client is used to interact with experimental Kubernetes features.
+type $.Group$Client struct {
+	*RESTClient
+}
+`
+
+var nameSpacerImplTemplate = `
+func (c *$.Group$Client) $.type.Name.Name$s(namespace string) $.type.Name.Name$Interface {
+	return new$.type.Name.Name$s(c, namespace)
+}
+`
+
+var newClientTemplate = `
+// New$.Group$ creates a new $.Group$Client for the given config.
+func New$.Group$(c *Config) (*$.Group$Client, error) {
+	config := *c
+	if err := set$.Group$Defaults(&config); err != nil {
+		return nil, err
+	}
+	client, err := RESTClientFor(&config)
```

Comment:
@wojtek-t, I think we need to let the framework track functions (here it's `RESTClientFor`) as well, so that we can track import for functions. To do so, we need to 
1. add a member variable `Functions` in types.Package, https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/types/types.go#L72
2. add the functions to Universe during parsing, https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/parser/parse.go#L295

I'll send a PR. WDYT?